### ISSUE TITLE
False negative for invalid indent of a flow node to the parent block node

### ISSUE THREAD

**Issue by daipom** [adrienverge/yamllint#833]
Example:

```yaml
foo: [
  "aaa"
]
```

```console
$ yamllint --version
yamllint 1.38.0
$ yamllint /tmp/foo.yaml
/tmp/foo.yaml
  1:1       warning  missing document start "---"  (document-start)
```

I expect an error like the following detected:

```
  3:1 error wrong indentation: ...
```

Ref:

- https://yaml.org/spec/1.2.2/#823-block-nodes
  - > Flow nodes must be indented by at least one more space than the parent block collection.

**Comment by andrewimeson** [comment]
Is your contention that there's a style issue, or that it's not a valid flow-style sequence (array)?

```console
$ cat << 'EOF' | python3 -c "import sys, yaml, json; json.dump(yaml.safe_load(sys.stdin), sys.stdout)"

foo: [
  "aaa"
]
EOF

{"foo": ["aaa"]}

**Comment by daipom** [comment]
Strictly speaking, it should violates the YAML specification, although some parser implementations can parse this.

> https://yaml.org/spec/1.2.2/#823-block-nodes
> 
>     * > Flow nodes must be indented by at least one more space than the parent block collection.

**Comment by daipom** [comment]
For example, [js-yaml](https://github.com/nodeca/js-yaml) cannot parse it.

https://nodeca.github.io/js-yaml/#yaml=LS0tCmZvbzogWwogICJiYXIiCl0K
